src/inmanta/data/dataview.py

```diff
@@ -486,68 +508,70 @@ def get_extra_url_parameters(self) -> dict[str, str]:
-
-        query_builder = SimpleQueryBuilder(
+        new_query_builder = SimpleQueryBuilder(
             select_clause="SELECT *",
             prelude=f"""
-                /* the recursive CTE is the second one, but it has to be specified after 'WITH' if any of them are recursive */
-                /* The latest_version CTE finds the maximum released version number in the environment */
-                WITH RECURSIVE latest_version AS (
+               WITH latest_version AS (
                     SELECT MAX(public.configurationmodel.version) as version
                     FROM public.configurationmodel
                     WHERE public.configurationmodel.released=TRUE AND environment=$1
-                ),
-                /*
-                emulate a loose (or skip) index scan (https://wiki.postgresql.org/wiki/Loose_indexscan):
-                1 resource_id at a time, select the latest (released) version it exists in.
-                */
-                cte AS (
-                    /* Initial row for recursion: select relevant version for first resource */
-                    ( {subquery_latest_version_for_single_resource(higher_than=None)} )
-                    UNION ALL
-                    SELECT next_r.*
-                    FROM cte curr_r
-                    CROSS JOIN LATERAL (
-                        /* Recurse: select relevant version for next resource (one higher in the sort order than current) */
-                        {subquery_latest_version_for_single_resource(higher_than="curr_r.resource_id")}
-                    ) next_r
+                ), versioned_resources AS (
+                    SELECT
+                        rps.*,
+                        CASE
+                            -- TODO: first branch may be redundant
+                            WHEN EXISTS (
+                                SELECT 1
+                                FROM resource AS r
+                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id AND r.model = (
+                                    SELECT version FROM latest_version
+                                )
+                            ) THEN (SELECT version FROM latest_version)
+                            ELSE (
+                                SELECT MAX(r.model)
+                                FROM resource AS r
+                                JOIN configurationmodel AS m
+                                    ON r.environment = m.environment AND r.model = m.version AND m.released = TRUE
+                                WHERE r.environment = rps.environment AND r.resource_id = rps.resource_id
+                                GROUP BY r.resource_id
+                            )
+                        END AS version
+                    FROM resource_persistent_state AS rps
+                    WHERE rps.environment = $1
+                ), result AS (
+                    SELECT
+                        r.resource_id,
```

Comment:
```suggestion
                        rps.resource_id,
```
I believe this should make some edge cases more performant.